pettingzoo/utils/env.py

```diff
@@ -3,12 +3,12 @@
 import warnings
 from typing import Any, Dict, Iterable, Iterator, List, Optional, Tuple, TypeVar
 
-import gym
-import numpy as np
+import gym  # type: ignore
+import numpy as np  # type: ignore
 
 ObsType = TypeVar("ObsType")
 ActionType = TypeVar("ActionType")
-AgentID = str
+AgentID = Optional[str]
```

Comment:
I'm not too familiar with dealing with typings, but is there a reason why this is Optional and not just a str?

Reply:
`Optional` means `AgentID` is also allowed to be `None`. I don't have a lot of experience with typing either but if I don't put `Optional`, I get the following error:
```
./PettingZoo/pettingzoo/utils/env.py:239:22 - error: Cannot assign member "agent_selection" for type "AECEnv"
    Expression of type "AgentID | None" cannot be assigned to member "agent_selection" of class "AECEnv"
      Type "AgentID | None" cannot be assigned to type "AgentID"
        Type "None" cannot be assigned to type "AgentID" (reportGeneralTypeIssues)
```